pyhap/characteristic.py

```diff
@@ -221,10 +236,10 @@ def to_HAP(self):
             hap_rep.update({k: self.properties[k] for k in
                             self.properties.keys() & PROP_NUMERIC})
         elif self.properties[PROP_FORMAT] == HAP_FORMAT_STRING:
-            if len(self.value) > 64:
-                hap_rep[HAP_REPR_MAX_LEN] = min(len(self.value), 256)
+            if len(self.get_value()) > 64:
```

Comment:
Maybe do `value = self.get_value()` at the beginning of the method, otherwise you could potentially call the callback two times in this if

Reply:
Yup, thanks for checking. Pushed changes.